File: hw/aimacana/u2/hw18mongodb/python/main.py
import sys
from pathlib import Path

# Add the project root to sys.path so modules can import from model/controller/database/view correctly
root_dir = str(Path(__file__).resolve().parent)
if root_dir not in sys.path:
    sys.path.append(root_dir)

from database.connection import MongoDBConnection
from controller.ingredient_controller import IngredientController
from controller.product_controller import ProductController
from view.mongodb_view import MongoDBView

def main():
    try:
        db = MongoDBConnection.get_database()
        
        ingredient_controller = IngredientController(db)
        product_controller = ProductController(db)
        
        view = MongoDBView(ingredient_controller, product_controller)
        view.run_demo()
        
    except Exception as e:
        print(f"\n[FATAL ERROR] Application failed to start: {e}", file=sys.stderr)
    finally:
        MongoDBConnection.close()

if __name__ == "__main__":
    main()

File: hw/aimacana/u2/hw18mongodb/typescript/src/main.ts
import { MongoDBConnection } from './database/connection.js';
import { IngredientController } from './controller/ingredientController.js';
import { ProductController } from './controller/productController.js';
import { MongoDBView } from './view/mongodbView.js';

async function main() {
    try {
        const db = await MongoDBConnection.getDatabase();

        const ingredientController = new IngredientController(db);
        const productController = new ProductController(db);

        const view = new MongoDBView(ingredientController, productController);
        await view.runDemo();

    } catch (error) {
        console.error('\n[FATAL ERROR] Application failed to start:', error);
    } finally {
        await MongoDBConnection.close();
    }
}

main();
